app/scripts/directives/apropiaciones/lista_apropiaciones.js

```diff
@@ -1,161 +1,188 @@
-                        if (response.data.Body !== null) {
-                            response.data.Body = response.data.Body.filter(function (a) {
-                                // funcion para filtrar rubros por codigo 
-                                return a.Codigo.startsWith($scope.tipo);
-                            });
-                            self.gridOptions.data = response.data.Body .sort(function (a, b) {
-                                if (a.Codigo < b.Codigo) { return -1; }
-                                if (a.Codigo > b.Codigo) { return 1; }
-                                return 0;
-                            });
-                            self.max_level = 0;
-                            var level = 0;
-                            // for (var i = 0; i < self.gridOptions.data.length; i += 1) {
-                            //     level = (self.gridOptions.data[i].Codigo.match(/-/g) || []).length;
-                            //     if (level > self.max_level) {
-                            //         self.max_level = level;
-                            //     }
-                            // }
-
-                            // for (var j = 0; j < self.gridOptions.data.length; j += 1) {
-                            //     level = (self.gridOptions.data[j].Codigo.match(/-/g) || []).length;
-                            //     if (level < self.max_level) {
-                            //         self.gridOptions.data[j].$$treeLevel = level;
-                            //     }
-                            // }
-
-                        } else {
-                            self.gridOptions.data = [];
-                        }
-                    });
-                };
-
-
-
-                self.gridOptions.onRegisterApi = function (gridApi) {
-                    //set gridApi on scope
-                    self.gridApi = gridApi;
-                    self.gridApi.grid.registerDataChangeCallback(function () {
-                        self.gridApi.treeBase.expandAllRows();
-                    });
-                    self.gridApi.selection.on.rowSelectionChanged($scope, function (row) {
-                        $scope.apropiacion = row.entity;
-                    });
-                };
-
-                self.gridOptions.isRowSelectable = function (row) {
-                    if (row.treeNode.children.length > 0 && $scope.selhijos === true) {
-                        return true;
-                    } else {
-                        return true;
-                    }
-                };
-
-
-                self.gridOptions.multiSelect = false;
+        $scope.$watchGroup(
+          ["unidadejecutora", "tipofinanciacion"],
+          function () {
+            if (
+              $scope.unidadejecutora != undefined &&
+              $scope.tipofinanciacion != undefined
```

Comment:
Dejar como estaba antes:
```suggestion
              $scope.unidadejecutora !== undefined &&
              $scope.tipofinanciacion !== undefined
```
... o por qué se cambió?

Reply:
Se cambió para hacer pruebas sobre la validación del árbol con el año 2021, sin embargo, se deja el modo estricto de comparación